libraries/AP_Notify/DiscoLED.h

```diff
@@ -0,0 +1,39 @@
+/*
+  Copyright (C) 2016 Mathieu Othacehe. All rights reserved.
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+#include <AP_HAL_Linux/PWM_Sysfs.h>
+#include "RGBLed.h"
+
+class DiscoLED: public RGBLed
+{
+public:
+    DiscoLED();
+    bool hw_init(void);
+    bool hw_set_rgb(uint8_t r, uint8_t g, uint8_t b);
+private:
+    Linux::PWM_Sysfs_Bebop *red_pwm;
+    Linux::PWM_Sysfs_Bebop *green_pwm;
+    Linux::PWM_Sysfs_Bebop *blue_pwm;
```

Comment:
Lucas,

My understanding here is that when we are constructing global Plane object, we have the following chain of constructors :

Plane -> AP_Notify -> Disco_LED -> PWM_Sysfs_Base

And then, in PWM_Sysfs_Base, we are not guaranteed that hal reference has been initialized. So I did this PR, but I'm not sure this is the right thing to do:

https://github.com/ArduPilot/ardupilot/pull/5076